### PR TITLE
Implemented add lesson

### PR DESCRIPTION
**Pull request by H-horizon**
Implemented add lesson lesson
The delimiter is now ;;

### CODE REVIEW

File: src/main/java/seedu/duke/common/Messages.java [hunk 1]
```diff
@@ -1,4 +1,5 @@
 package seedu.duke.common;
 
 public class Messages {
+    public static final String ADDED_LESSON_MESSAGE = "Added: ";
```
Comment (8kdesign) [inline src/main/java/seedu/duke/common/Messages.java:4]: Might want to call it MESSAGE_ADDED_LESSON instead, so all messages will have the same prefix.

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+    public static final String DELIMITER = ";;";
+    public static final int NUMBER_OF_ARGUMENTS = 5;
+    public static final String TUTORIAL = "TUTORIAL";
+    public static final String LECTURE = "LECTURE";
+    public static final String LAB = "LAB";
+    public static final int LESSON_TYPE_INDEX = 0;
+    public static final int TIME_INDEX = 1;
+    public static final int ONLINE_LINK_INDEX = 2;
+    public static final int TEACHING_STAFF_NAME_INDEX = 3;
+    public static final int TEACHING_STAFF_EMAIL_INDEX = 4;
```
Comment (8kdesign) [inline src/main/java/seedu/duke/lesson/Lesson.java:18]: These can be shifted to a Constants class in the common package when we merge.

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 3]
```diff
@@ -38,4 +59,43 @@ public void setTeachingStaff(TeachingStaff teachingStaff) {
...
+
+        case LAB:
+            return LessonType.LAB;
+
+        default:
+            //throw required exception if parser didn't handle
+            System.out.println("Wrong name");
+            return LessonType.LECTURE; //temporary
+        }
+    }
```
Comment (8kdesign) [inline src/main/java/seedu/duke/lesson/Lesson.java:79]: I have this part in the storage code too. Maybe we can store this method in a class in the common package.

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 4]
```diff
@@ -38,4 +59,43 @@ public void setTeachingStaff(TeachingStaff teachingStaff) {
...
+
+    /**
+     * User input: "add tutorial ;; Wednesday 9 am - 10am ;; https://nus-sg.zoom.us/j/abc"
+     * Parser input: "tutorial ;; Wednesday 9 am - 10am ;; https://nus-sg.zoom.us/j/abc ;;  ;; "
+     * Parser will input " " and delimiters in parser input equal to five i,e
+     * Tentatively discussed with Ivan(REMOVE THIS COMMENT AFTER IMPLEMENTATION)
+     * This method add a new lesson to the list of lessons of a module
```
Comment (8kdesign) [inline src/main/java/seedu/duke/lesson/Lesson.java:86]: Might need to modify this to meet the Java Coding Standards.

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 5]
```diff
@@ -38,4 +59,43 @@ public void setTeachingStaff(TeachingStaff teachingStaff) {
...
+     *
+     * @param parserInput is the input obtained from parser assumed to be in the correct format
+     * @param lessonList  is the list of lessons for one module
+     */
+    public static void addNewLesson(String parserInput, ArrayList<Lesson> lessonList) {
+        String[] arguments = getArguments(parserInput);
+        Lesson newLesson = new Lesson(arguments);
+        lessonList.add(newLesson);
+        UI.printAddNewLesson(newLesson.lessonType);
+    }
```
Comment (8kdesign) [inline src/main/java/seedu/duke/lesson/Lesson.java:96]: Would be good if you can extract this into a class in the commands package.

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 6]
```diff
@@ -38,4 +59,43 @@ public void setTeachingStaff(TeachingStaff teachingStaff) {
...
+
+    private static String[] getArguments(String parserInput) {
+        return parserInput.split(DELIMITER, NUMBER_OF_ARGUMENTS);
+    }
```
Comment (8kdesign) [inline src/main/java/seedu/duke/lesson/Lesson.java:100]: Will the validity of arguements be checked here? Or will that be done in the parser?

**Review by 8kdesign — CHANGES_REQUESTED** [review]
Our part shares some constants. Maybe we can share them via a Constants class in commons. Might need to extract out the methods into seperate classes in the commands package so that the parser can pass the command to duke to execute. Also, I notice you are missing the remove function for lessons.

File: src/main/java/seedu/duke/lesson/Lesson.java (hunk 2 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/lesson/Lesson.java:18]: Was planning on doing so after I'm done implementing all commands. Thank you for the suggestion anw.

File: src/main/java/seedu/duke/lesson/Lesson.java (hunk 3 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/lesson/Lesson.java:79]: I'll have a look. Sure idm

File: src/main/java/seedu/duke/lesson/Lesson.java (hunk 4 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/lesson/Lesson.java:86]: This is not the final comment. I wrote this for Ivan. Will change it after he's done implementing the parser. No worries

File: src/main/java/seedu/duke/lesson/Lesson.java (hunk 6 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/lesson/Lesson.java:100]: Parser

File: src/main/java/seedu/duke/lesson/Lesson.java (hunk 5 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/lesson/Lesson.java:96]: I wanted to wait for the merging to discuss the names and refactor the code. I believe every lesson commands should be kept in the command package. Just didn't to create new classes in the skeleton rn.

File: src/main/java/seedu/duke/common/Messages.java (hunk 1 above)
Comment (H-horizon) [reply src/main/java/seedu/duke/common/Messages.java:4]: Will change it the constant name

**Comment by H-horizon** [comment]
> Also, I notice you are missing the remove function for lessons.

This branch contains only the methods needed to add a lesson. The methods for other commands will be added from their respective branches.

Thank you Wen Hao!!

**Pull request closed without merging**